feature_engine/prediction/target_mean_regressor.py

```diff
@@ -0,0 +1,124 @@
+    variables: list, default=None
+        The list of input variables. If None, the estimator will use all variables as
+        input features (except datetime).
+
+    bins: int, default=5
+        If the dataset contains numerical variables, the number of bins into which
+        the values will be sorted.
+
+    strategy: str, default='equal_width'
+        Whether the bins should of equal width ('equal_width') or equal frequency
+        ('equal_frequency').
+
+    Attributes
+    ----------
+    variables_categorical_:
+        The group of categorical input variables that will be used for prediction.
+
+    variables_numerical_:
+        The group of numerical input variables that will be used for prediction.
+
+    pipeline_:
+        A Sickit-learn Pipeline with a dicretiser and/or encoder. Used to determine the
+        mean target value per category or bin, per variable.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Learn the mean target value per category or per bin, for each variable.
+    predict:
+        Predict using the average of the target mean value across variables.
+    score:
+        Return the coefficient of determination of the prediction.
+
+    See Also
+    --------
+    feature_engine.encoding.MeanEncoder
+    feature_engine.discretisation.EqualWidthDiscretiser
+    feature_engine.discretisation.EqualFrequencyDiscretiser
+
+    References
+    ----------
+    Adapted from:
+
+    .. [1] Miller, et al. "Predicting customer behaviour: The University of Melbourne’s
+        KDD Cup report". JMLR Workshop and Conference Proceeding. KDD 2009
+        http://proceedings.mlr.press/v7/miller09/miller09.pdf
+    """
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Learn the mean target value per category or bin.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+
+        y : pandas series of shape = [n_samples,]
+            The target variable.
+        """
+
+        if type_of_target(y) == "binary":
+            raise ValueError(
+                "Trying to fit a regression to a binary target is not "
+                "allowed by this transformer. Check the target values "
+                "or set regression to False."
+            )
+
+        return super().fit(X, y)
+
+    def predict(self, X: pd.DataFrame) -> pd.Series:
+        """
+        Predict using the average of the target mean value across variables.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, ]
+            The input samples.
+
+        Return
+        -------
+        y_pred: ndarray of shape (n_samples,)
+            Returns predicted values.
+        """
+
+        y_pred = self._predict(X)
+        if not isinstance(y_pred, np.ndarray) or y_pred.ndim != 1:
+            raise ValueError(
```

Comment:
we normally raise errors to help users understand what they are doing wrong. 

But here, we are raising an error that basically checks that our method _predict() is doing the right job. So if _predict() does not, the user won't have any clue what to do. And won't be able to fix it, other than messing with the source code. So it is not useful.

Instead, we need to ensure the output of _predict, is what it should be.

Why did you add this check all along? what did I miss?